File: spring-data-requery-java/src/test/java/com/coupang/springframework/data/requery/java/domain/sample/AbstractAccount.java
package com.coupang.springframework.data.requery.java.domain.sample;

import com.coupang.springframework.data.requery.domain.AbstractPersistable;
import io.requery.Column;
import io.requery.Entity;
import io.requery.Key;
import lombok.Getter;
import lombok.Setter;

/**
 * AbstractAccount
 *
 * @author debop@coupang.com
 * @since 18. 6. 14
 */
@Getter
@Setter
@Entity(name = "Accounts")
public class AbstractAccount extends AbstractPersistable<Long> {

    private static final long serialVersionUID = 3616603959214418196L;

    @Key
    protected Long id;


    @Column(name = "account_name")
    protected String name;


}

File: spring-data-requery-kotlin/src/test/kotlin/com/coupang/springframework/data/requery/domain/sample/AbstractAccount.kt
package com.coupang.springframework.data.requery.domain.sample

import com.coupang.springframework.data.requery.domain.AbstractPersistable
import io.requery.Column
import io.requery.Entity
import io.requery.Generated
import io.requery.Key

/**
 * com.coupang.springframework.data.requery.domain.sample.AbstractAccount
 * @author debop
 * @since 18. 5. 30
 */
@Entity
abstract class AbstractAccount: AbstractPersistable<Long>() {

    @get:Key
    @get:Generated
    abstract val id: Long?


    @get:Column(name = "account_name", length = 48)
    abstract var name: String?
}